server/service/osquery.go

```diff
@@ -818,14 +818,19 @@ func (svc *Service) SubmitDistributedQueryResults(
 				ll = level.Error(svc.logger)
 			}
 			ll.Log("query", query, "message", messages[query], "hostID", host.ID)
+		}
 
+		if failed {
+			// if a query failed, don't try to ingest it
+			continue
```

Comment:
The idea of not even calling the ingestion func if the query failed sounds good to me, we were inconsistent in how it was handled anyway, but I see that in some cases we did log that failure. It did generate a lot of noise, but don't we want to log better detailed information than what we did? Or do we consider that the logging at line 820 above is doing that job?

Reply:
oh you're right! Yeah, I'll change it around so we just skip the section below instead of the whole thing.

Reply:
Actually... those errors are _ingestion_ errors, this would be a query execution error, which we are logging... so I think I stand by this original way.

Reply:
If a query failed to execute in the host, we'd like to know why (to ease troubleshooting). That said.

1. IIRC, the silencing of "no such table" errors was added before we added support for discovery queries.
Maybe now we should also log those?
2. Should we log `level=error` instead? (Otherwise users have to set debug mode and restart the server)

So we could replace the two `if`s with:
```go
if failed {
  level.Error(svc.logger).Log("query", query, "message", messages[query], "hostID", host.ID, "platform", host.Platform, "status", status)
  // if a query failed, don't try to ingest it
  continue
}
```

Am all ears.

Reply:
As it is right now, we will be logging as debug unless it's a denylisted:

```
if failed && messages[query] != "" && !noSuchTableRegexp.MatchString(messages[query]) {
			ll := level.Debug(svc.logger)
			// We'd like to log these as error for troubleshooting and improving of distributed queries.
			if messages[query] == "distributed query is denylisted" {
				ll = level.Error(svc.logger)
			}
			ll.Log("query", query, "message", messages[query], "hostID", host.ID)

		}
		
```

I think that's ok, because it's not noisy by default. We'll have to improve this later on as part of another issue to log these things but only once every so often instead of once per host in the worse case.

Reply:
OK, happy to discuss improvements on a separate PR.